packages/pillarbox-playlist/src/components/modal/pillarbox-previous-item-button.js

```diff
@@ -0,0 +1,51 @@
+import videojs from 'video.js';
+
+/**
+ * @ignore
+ * @type {typeof import('video.js/dist/types/button').default}
+ */
+const Button = videojs.getComponent('Button');
+
+/**
+ * The previous item button for the playlist ui. When clicked moves to the
+ * previous item in the playlist.
+ */
+class PillarboxPlaylistPreviousItemButton extends Button {
+  constructor(player, options) {
+    super(player, options);
+    this.handleLanguagechange();
+    this.setIcon('previous-item');
+  }
+
+  ready() {
+    this.$('.vjs-icon-placeholder').classList.toggle(`vjs-icon-previous-item`, true);
+  }
+
+  /**
+   * Get the playlist instance associated with the player.
+   *
+   * @returns {import('pillarbox-playlist.js').default} The playlist instance.
+   */
+  playlist() {
+    return this.player().pillarboxPlaylist();
+  }
```

Comment:
Could you explain the intention behind this? Why not have a parent component and benefit from inheritance?

Reply:
I would have to create an intermediate class for each kind of component this plugin, or use some dynamic mixin strategy which is not great. I settled for this solution since it's just a convenience function.